util/request_source.go

```diff
@@ -52,6 +55,12 @@ func (r *RequestSource) SetRequestSourceType(tp string) {
 	r.RequestSourceType = tp
 }
 
+// IsBackground checks whether the request is a background job.
+func (r *RequestSource) IsBackground() bool {
+	// TODO: supports set as background by user in the future.
+	return r.RequestSourceType == InternalTxnTools
```

Comment:
I wander what scenario it is to expected pass this `tools` source-type, as most ecosystem tools are not running in txn mode. BTW, tidb-lightning's tidb backend is running with normal txn, but since it's executed via normal sql statements, it's request source is unlikely to be tools either.

Reply:
@glorv  Do you think adding a type in these 2 places is a comprehensive way? https://github.com/pingcap/tidb/pull/44628/files

Reply:
I'd like to add a `IsBackground` field the same way as the `ResourceGroupName` field. Determine the value of background implicitly by `RequestSourceType` is a bit weird to me.

Reply:
make sense to me.